plugin/trino-hive/src/test/java/io/trino/plugin/hive/TestHiveDecimalToVarcharCoercion.java

```diff
@@ -0,0 +1,107 @@
+
+    @Test
+    public void testLongDecimalToVarcharCoercion()
+    {
+        String tableName = "test_long_decimal_to_varchar_coercion_" + randomTableSuffix();
+        assertUpdate("CREATE TABLE " + tableName + " (col1 DECIMAL(23,3), col2 int, col3 int) with ( FORMAT = 'PARQUET', partitioned_by = ARRAY['col3'])");
+        assertUpdate("INSERT INTO " + tableName + "(col1, col2, col3) VALUES(DECIMAL '100.13', 1, 1)", 1);
+        query("ALTER TABLE " + tableName + " DROP COLUMN col1");
+        query("ALTER TABLE " + tableName + " ADD COLUMN col1 VARCHAR");
+        assertUpdate("INSERT INTO " + tableName + " (col1, col2, col3) VALUES('1000.222', 2, 2)", 1);
+        assertQuery("SELECT col1, col2, col3 FROM " + tableName, "VALUES ('100.130', 1, 1), ('1000.222', 2, 2)");
+    }
+
+    @Test
+    public void testShortDecimalWithTooLongScaleToVarcharCoercion()
+    {
+        String tableName = "test_long_decimal_to_varchar_coercion_" + randomTableSuffix();
+        assertUpdate("CREATE TABLE " + tableName + " (col1 DECIMAL(5,3), col2 int, col3 int) with ( FORMAT = 'PARQUET', partitioned_by = ARRAY['col3'])");
+        assertUpdate("INSERT INTO " + tableName + "(col1, col2, col3) VALUES(DECIMAL '10.1333', 1, 1)", 1);
+        query("ALTER TABLE " + tableName + " DROP COLUMN col1");
+        query("ALTER TABLE " + tableName + " ADD COLUMN col1 VARCHAR");
+        assertUpdate("INSERT INTO " + tableName + " (col1, col2, col3) VALUES('1000.2222', 2, 2)", 1);
+        assertQuery("SELECT col1, col2, col3 FROM " + tableName, "VALUES ('10.133', 1, 1), ('1000.2222', 2, 2)");
+    }
+
+    @Test
+    public void testShortDecimalToBoundedVarcharCoercion()
+    {
+        String tableName = "test_short_decimal_to_bounded_varchar_coercion_" + randomTableSuffix();
+        assertUpdate("CREATE TABLE " + tableName + " (col1 DECIMAL(10,3), col2 int, col3 int) with ( FORMAT = 'PARQUET', partitioned_by = ARRAY['col3'])");
+        assertUpdate("INSERT INTO " + tableName + "(col1, col2, col3) VALUES(DECIMAL '100.13', 1, 1)", 1);
+        query("ALTER TABLE " + tableName + " DROP COLUMN col1");
+        query("ALTER TABLE " + tableName + " ADD COLUMN col1 VARCHAR(5)");
+        assertUpdate("INSERT INTO " + tableName + " (col1, col2, col3) VALUES('100.2', 2, 2)", 1);
+        assertQuery("SELECT col1, col2, col3 FROM " + tableName, "VALUES ('100.1', 1, 1), ('100.2', 2, 2)");
+    }
+
+    @Test
+    public void testLongDecimalToBoundedVarcharCoercion()
+    {
+        String tableName = "test_short_decimal_to_bounded_varchar_coercion_" + randomTableSuffix();
+        assertUpdate("CREATE TABLE " + tableName + " (col1 DECIMAL(23,3), col2 int, col3 int) with ( FORMAT = 'PARQUET', partitioned_by = ARRAY['col3'])");
+        assertUpdate("INSERT INTO " + tableName + "(col1, col2, col3) VALUES(DECIMAL '100.13', 1, 1)", 1);
+        query("ALTER TABLE " + tableName + " DROP COLUMN col1");
+        query("ALTER TABLE " + tableName + " ADD COLUMN col1 VARCHAR(5)");
```

Comment:
Some file formats (eg CSV) are positional. Dropping column other than the last and re-adding it may have weird results (like shifting data).

Let's maybe make the interesting column the last one.

Also, do we need 3 columns, or would 2 columns be enough?
Can the columns have names more distinct than col1, col2, col3?

like `dummy integer, evolved_column decimal(...)` ?

Reply:
I think I will drop this class entirely, these test cases are covered by product test anyway

Reply:
and BTW it cannot be the last one because the last one is a partitioning column, and partitioning column has to be the last one.

Reply:
> I think I will drop this class entirely

let me know whether you want to drop it or should i continue with review of it

Reply:
I left it so you can review it ;)
I am trying to add other file formats here but some don't work because of the problem with dropping/adding columns

Reply:
Since Trino doesn't have proper support for changing column type, let's run with product tests.

Reply:
I've just wanted to have a test that will check that coercion to too small varchar will fail with expected exception.

Reply:
i think it's fine not to have it